assets/src/edit-story/components/panels/panel/panel.js

```diff
@@ -81,9 +82,11 @@ function Panel({
     return persisted?.isCollapsed ?? collapsedByDefault;
   });
   const [expandToHeight, setExpandToHeight] = useState(
-    persisted?.expandToHeight || initialHeight
+    Math.min(persisted?.expandToHeight, maxHeight) || initialHeight
+  );
+  const [height, setHeight] = useState(
+    Math.min(persisted?.height, maxHeight) || initialHeight
```

Comment:
`maxHeight` can override the persisted height if it is less than what is saved in local storage